client/thin-replica-client/src/thin_replica_client.cpp

```diff
@@ -573,6 +573,7 @@ void ThinReplicaClient::receiveUpdates() {
       if (servers_with_hash >= (2 * config_->max_faulty + 1) && !hash_has_agreement) {
         std::string msg{"Internal Error: Couldn't find agreement, 2f+1 replicas generated mismatching hashes"};
         LOG_ERROR(logger_, msg);
+        stop_subscription_thread_ = true;
```

Comment:
Instead of setting it before every `throw`, we can set this variable inside the wrapper's catch statement.

Reply:
Updated.